p2pserver/message/utils/msg_handler.go

```diff
@@ -208,9 +212,12 @@ func TransactionHandle(data *msgTypes.MsgPayload, p2p p2p.P2P, pid *evtActor.PID
 	log.Trace("[p2p]receive transaction message", data.Addr, data.Id)
 
 	var trn = data.Payload.(*msgTypes.Trn)
-	actor.AddTransaction(trn.Txn)
-	log.Trace("[p2p]receive Transaction message hash", trn.Txn.Hash())
-
+	if !txCache.Contains(trn.Txn) {
```

Comment:
storing tx is totally wrong.  
1. note `trn.Txn` is a pointer, same transactions received from p2p will allocated in different memory allocation. so `Contains` will always return `false`.  
2. and it will cause dos attack since 100000*1M(per tx) = 100G.

replace it with tx hash.

Reply:
I change storing tx to storing txhash